### DISCUSSION TITLE
Defining field uniqueness, or ways of hiding/changing error messages

### DISCUSSION THREAD

**Discussion by demitchell14** [Budibase/budibase#5143] [Help] [answered]
Hello,

I'm trying to figure out Budibase, and one thing that I need to be able to do is require a field within the database to be unique.. In my case, it's a serial number.

I couldn't figure out a way to do this within the budibase DB, so I installed postgres and planned on using this... well, there was no easy way within the dashboard to mark a field as unique, so I did it manually and changed the table directly within postgres.

This works just fine and I get an error whenever I try to enter a duplicate serial, however, the error is literally a database error.

Does anyone know of a way around this problem, or if there is a way to define error messages?

Also... Is there a way to stop actions from continuing down the action chain?

**Comment by melohagan** [comment, marked as answer]
So at the moment, the only action that stops the chain is **Validate Form**

This can be used, in combination with App State and a Data Provider to enforce uniqueness on the Budibase DB, via form entry.

1. Add a data provider to your form:
![Screenshot 2022-03-29 at 10 28 01](https://user-images.githubusercontent.com/101575380/160580404-d1b33804-5b31-4e03-99c2-1f36f8d41df5.png)

2. Define actions on the 'Save Button'. By default there should be 3 actions already: Validate Form, Save Row, and Navigate To. 
    Click 'Add action' > 'Update State'. Move it to be the first action.
    **Key**: MatchOnName
    **Value**: 
    ```
    let rows = $("Data Provider.Rows");
    let uniqueName = $("Form.Fields.Name");
    return rows.some(row => row.Name?.toLowerCase() === uniqueName?.toLowerCase()) ? uniqueName : undefined;
    ```
    This will allow us to know when validating if the Name field is unique (ignoring case).

3. Click on the Name field > 'Configure Validation'
    Enter the following:
    ![Screenshot 2022-03-29 at 10 33 56](https://user-images.githubusercontent.com/101575380/160581476-2d5411fd-6bf9-42dc-83b5-4a3574f7839f.png)

In the near future, you will also be able to add a 'Continue If' action step for more flexibility, as outlined in this issue: https://github.com/Budibase/budibase/issues/4143

**Reply by demitchell14** [reply]
Awesome, I actually figured out that you can use conditional validation, which is what I ultimately did. Thanks!

I look forward to the 'Continue If' action though
